app/routes/circulars/index.tsx

```diff
@@ -214,7 +214,7 @@ export default function () {
           <ol>
             {allItems.map(({ circularId, subject }) => (
               <li key={circularId} value={circularId}>
-                <Link to={`/circulars/${circularId}`}>{subject}</Link>
+                <Link to={`/circulars/${circularId}?query=${query ?? ''}`}>{subject}</Link>
```

Comment:
- [ ] We need to save both the query string and page number.
- [ ] I suggest that we lift the generation of the query string out of the loop, to avoid repeatedly evaluating it on every iteration. Like this:

```ts
  const [searchParams] = useSearchParams()
  let searchParamsString = searchParams.toString()
  if (searchParamsString) searchParamsString = `?${searchParamsString}`
```

Then, inside the loop:

```jsx
<Link to={`/circulars/${circularId}${searchParamsString}`}>
```

Reply:
ah, I neglected to consider the page number, thanks. I'll change the query generation as you suggest as well.